vsphere/resource_vsphere_vnic.go

```diff
@@ -0,0 +1,517 @@
+package vsphere
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-vsphere/vsphere/internal/helper/hostsystem"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/types"
+	"log"
+	"strconv"
+	"strings"
+)
+
+func resourceVsphereNic() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVsphereNicCreate,
+		Read:   resourceVsphereNicRead,
+		Update: resourceVsphereNicUpdate,
+		Delete: resourceVsphereNicDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: vNicSchema(),
+	}
+}
+
+func vNicSchema() map[string]*schema.Schema {
+	base := BaseVMKernelSchema()
+	base["host"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "ESX host the interface belongs to",
+		ForceNew:    true,
+	}
+
+	return base
+}
+
+func resourceVsphereNicRead(d *schema.ResourceData, meta interface{}) error {
+	ctx := context.TODO()
+	client := meta.(*VSphereClient).vimClient
+	tfNicID := d.Id()
+
+	toks := strings.Split(tfNicID, "_")
```

Comment:
I like to just use d.Id() directly here. That way you aren't storing the ID in a separate var again that is only used once.